### PR TITLE
feat(client): Adds methods to set/get the client.

### PR DESCRIPTION
**Pull request by bobylito**
This is not really a feature for the users of the JS Helper, but more of
convenience for any library built on top of it.

See https://github.com/algolia/instantsearch.js/issues/1673

### CODE REVIEW

File: src/algoliasearch.helper.js [hunk 1]
```diff
@@ -1215,6 +1215,29 @@ AlgoliaSearchHelper.prototype.clearCache = function() {
...
 /**
+ * Updates the internal client instance. If the reference of the clients
+ * are equal then no update is actually done.
+ * @param  {AlgoliaSearch} newClient an AlgoliaSearch client
+ * @return {AlgoliaSearchHelper}
+ */
+AlgoliaSearchHelper.prototype.setClient = function(newClient) {
+  if (this.client === newClient) return this;
+
+  if (!newClient.addAlgoliaAgent) newClient.addAlgoliaAgent('JS Helper ' + version);
```
Comment (vvo) [inline src/algoliasearch.helper.js]: Should you call setClient from the constructor now that you have this init method?

File: test/integration-spec/helper.searchForFacetValues.js [hunk 2]
```diff
@@ -25,8 +25,7 @@ var dataset = [
 ];
 
 var config = {
-  searchableAttributes: ['f', 'f2'],
-  attributesForFaceting: ['f', 'f2']
+  attributesForFaceting: ['searchable(f)', 'searchable(f2)']
```
Comment (vvo) [inline test/integration-spec/helper.searchForFacetValues.js]: Is this related to the client change?

Comment (bobylito) [inline test/integration-spec/helper.searchForFacetValues.js]: it's related to the engine changing its behavior...

Comment (bobylito) [inline test/integration-spec/helper.searchForFacetValues.js]: thus breaking the test...

Comment (vvo) [inline test/integration-spec/helper.searchForFacetValues.js]: Got it!

File: src/algoliasearch.helper.js (hunk 1 above)
Comment (bobylito) [inline src/algoliasearch.helper.js]: Why not? I will try that!

Comment (bobylito) [inline src/algoliasearch.helper.js]: It works!

**Comment by bobylito** [comment]
All changes done @vvo thanks for the review
